### PR TITLE
fix tests on 1.7

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -11,6 +11,12 @@ if Sys.ARCH === :x86_64 || Sys.ARCH === :i686
     doctest(FixedPointNumbers, manual = false)
 end
 
+if !isdefined(Base, :checked_length)
```
Comment (johnnychen94) [inline test/runtests.jl:14]: Just a reminder comment for future clean-up

```suggestion
# Compat for Julia 1.7
if !isdefined(Base, :checked_length)
```

**Comment by kimikage** [comment]
I'm confused because `unsafe_length` is "unchecked" and `checked_length` is "checked".

File: test/fixed.jl [hunk 2]
```diff
@@ -647,7 +647,7 @@ end
     @test length(r) == 256
     QInt1 = Fixed{Int,1}
     @test length(QInt1(0):eps(QInt1):typemax(QInt1)-eps(QInt1)) == typemax(Int)
-    @test Base.unsafe_length(typemin(QInt1):eps(QInt1):typemax(QInt1)-eps(QInt1)) == -1
+    @test checked_length(typemin(QInt1):eps(QInt1):typemax(QInt1)-eps(QInt1)) == -1
```
Comment (kimikage) [inline test/fixed.jl:650]:
```suggestion
    @test_throws OverflowError length(typemin(QInt1):eps(QInt1):typemax(QInt1)-eps(QInt1))
```

File: test/normed.jl [hunk 3]
```diff
@@ -607,7 +607,7 @@ end
     NInt1 = Normed{UInt,1}
     @test length(NInt1(0):typemax(NInt1)-oneunit(NInt1)) == typemax(UInt)
     @test_throws OverflowError length(NInt1(0):typemax(NInt1))
-    @test Base.unsafe_length(NInt1(0):typemax(NInt1)) == 0  # overflow
+    @test checked_length(NInt1(0):typemax(NInt1)) == 0  # overflow
```
Comment (kimikage) [inline test/normed.jl:610]:
```suggestion
    @test_throws OverflowError length(NInt1(0):typemax(NInt1))
```

**Review by kimikage — COMMENTED** [review]
I think it is enough to simply check the overflow.

**Comment by KristofferC** [comment]
> Also, isn't 1.7 in the title a mistake for 1.8-DEV?

No, because the relevant changes are on the backport branch.

File: test/normed.jl [hunk 4]
```diff
@@ -607,7 +607,7 @@ end
     NInt1 = Normed{UInt,1}
     @test length(NInt1(0):typemax(NInt1)-oneunit(NInt1)) == typemax(UInt)
     @test_throws OverflowError length(NInt1(0):typemax(NInt1))
-    @test Base.unsafe_length(NInt1(0):typemax(NInt1)) == 0  # overflow
+    @test_throws OverflowError length(NInt1(0):typemax(NInt1))
```
Comment (kimikage) [inline test/normed.jl:610]: Ah..., we already had the same test above.  :sweat_smile:

**Comment by kimikage** [comment]
The segmentation fault on aarch64 nightly is strange, but at least it's not related to this PR.

**Pull request merged**